crates/sparrow-sources/src/in_memory.rs

```diff
@@ -0,0 +1,221 @@
+use std::sync::{Arc, RwLock};
+
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+use error_stack::{IntoReportCompat, ResultExt};
+use futures::{Stream, StreamExt, TryStreamExt};
+
+use sparrow_batch::Batch;
+use sparrow_core::ReadConfig;
+use sparrow_interfaces::Source;
+use sparrow_merge::old::homogeneous_merge;
+
+use crate::Error;
+
+/// Wraps the mutable in-memory batch container.
```

Comment:
I don't love describing something as `Wraps ...` since it means "to understand this you need to go understand that". It's the "See ..." as a dictionary definition.

Can we instead say -- `Shared, synchronized container for in-memroy batches` or something like that?

Reply:
That makes sense. updated